### ISSUE TITLE
Infinite recursion "'this' is expected an EventTarget object, but got [object Object]'

### ISSUE THREAD

**Issue by murillo128** [mysticatea/event-target-shim#12]
I am getting an error on my app, but it targets an infinite recursion on:

```
function getListeners(eventTarget) {
    const listeners = listenersMap.get(eventTarget);
    console.assert(listeners != null, "'this' is expected an EventTarget object, but got", eventTarget);
    return listeners || new Map()
}
```
At least on IE11, the console.assert when trying display my bogus `eventTarget` calls again the getListeners, which calls assert again.. and so on..

![image](https://user-images.githubusercontent.com/1070835/35225886-c87a663e-ff89-11e7-97d2-173794a02694.png)

**Comment by mysticatea** [comment]
Thank you for the report.

I confirmed it on IE11, one assertion generates many messages. Chrome and Firefox don't have the problem.

**Comment by mysticatea** [comment]
IE11 does not run it infinitely and it can check the cause object in their F12 tool.

I'm not sure if it's worth to remove the `"but got [object]"` part on IE even if people are going to not be able to check the cause object.

**Comment by murillo128** [comment]
IE runs recursively until heaps is exhausted and heap is aborted.  You can't either check the source object on dev tools, as when you try to watch the object, the properties are evaluated (i.e. if you have an onxxxx) and it runs the console.assert loop until you get another heap exhaustion.

IMHO, it would be much better just to throw an exception in this case.

**Comment by mysticatea** [comment]
I can check the object:

![image](https://user-images.githubusercontent.com/1937871/35272519-cc5886ec-0078-11e8-8480-7b291e2e16d2.png)

**Comment by murillo128** [comment]
I couldn't in my case, browser was completely unresponsive. 

I was working to expose the EventHandler interface to the windows object to pass the w3c web platform tools idl tests for a shim I was working on, which involved inspecting prototypes also.

**Comment by mysticatea** [comment]
Hmm, OK. I will fix it.

**Comment by murillo128** [comment]
Thank you , let me know if I can help in anything

**Comment by murillo128** [comment]
thanxs!

**Comment by mysticatea** [comment]
I published [v3.0.1](https://github.com/mysticatea/event-target-shim/releases/tag/v3.0.1). Thank you for the contribution!
